yt_dlp/extractor/pandatv.py

```
@@ -0,0 +1,68 @@
+    ExtractorError,
+    UserNotLive,
+    int_or_none,
+    traverse_obj,
+    unified_strdate,
+    url_or_none,
+    urlencode_postdata,
+)
+
+
+class PandatvLiveIE(InfoExtractor):
+    _VALID_URL = r'(?P<base_url>https?://(?:www\.|m\.)?pandalive\.co\.kr)/play/(?P<id>[\da-z]+)'
+    _TESTS = [{
+        'url': 'https://www.pandalive.co.kr/play/bebenim',
+        'info_dict': {
+            'id': 'bebenim',
+            'ext': 'mp4',
+            'channel': '릴리ෆ',
+            'title': r're:앙앙❤ \d{4}-\d{2}-\d{2} \d{2}:\d{2}',
+            'thumbnail': r're:https://cdn\.pandalive\.co\.kr/ivs/v1/.+/thumb.jpg',
+            'concurrent_view_count': int,
+            'like_count': int,
+            'live_status': 'is_live',
+            'upload_date': str,
+        },
+        'skip': 'The channel is not currently live',
+    }]
+
+    def _real_extract(self, url):
+        base_url, channel_id = self._match_valid_url(url).groups()
+        http_headers = {'Origin': base_url}
+
+        # Prepare POST data
+        post_data = {'action': 'watch', 'userId': channel_id, 'password': '', 'shareLinkType': ''}
+        post_data_bytes = urlencode_postdata(post_data)
+
+        # Fetch video metadata
+        video_meta = self._download_json(
+            'https://api.pandalive.co.kr/v1/live/play', channel_id, 'Downloading video meta data',
+            errnote=' Unable to download video meta data', data=post_data_bytes, expected_status=(200, 400))
+
+        # Check video metadata
+        if not video_meta.get('result'):
+            if traverse_obj(video_meta, ('errorData', 'code')) == 'castEnd':
+                raise UserNotLive(video_id=channel_id)
+            elif traverse_obj(video_meta, ('errorData', 'code')) == 'needAdult':
+                raise ExtractorError(
+                    'Adult verification is required. Check `--cookies` or `--cookies-from-browser` '
+                    'method in https://github.com/yt-dlp/yt-dlp#filesystem-options', expected=True)
```

- save the `error_code` so we can raise an unexpected error if it's something we're not aware of
- use `self.raise_login_required()` instead of writing your own cookies hint
```suggestion
        if not video_meta.get('result'):
            error_code = traverse_obj(video_meta, ('errorData', 'code', {str}))
            if error_code == 'castEnd':
                raise UserNotLive(video_id=channel_id)
            elif error_code == 'needAdult':
                self.raise_login_required('Adult verification is required for this stream')
            else:
                raise ExtractorError(f'API returned an error code: {error_code}')
```